### ISSUE TITLE
WebDAV: Not handling 401 for OPTIONS request?

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero-ios#409]
https://forums.zotero.org/discussion/93715/ios-sync-error-in-webdav-regression-error-build-169

Not tested, but this suggests we're not properly handling Basic Auth for OPTIONS requests?

**Comment by michalrentka** [comment]
I'm not able to verify this provider, since they only provide their service to universities it seems. But it worked for me with basic auth locally. Do we know about another webdav provider that uses basic auth so that I could test it there?

**Comment by dstillman** [comment]
Yeah, a local server works for me — there's a request for OPTIONS without a username, a 401 response, and then a follow-up request with credentials. Debug output logging doesn't seem to show the initial request, only one request with a 200 response.

What changed between 168 and 169 in terms of how we were sending credentials? Was 168 the hard-coded Basic Auth implementation, where we just formed the Authorization header ourselves?

**Comment by michalrentka** [comment]
Yes, 168 had the hardcoded header. Now I'm using Alamofire's `.authenticate` (https://github.com/Alamofire/Alamofire/blob/master/Documentation/Usage.md#authentication) which should support all auth methods required.

So I'm not quite sure what's wrong now. If we could find a provider that doesn't work and I could access that would help.

**Comment by dstillman** [comment]
https://forums.zotero.org/discussion/comment/397791/#Comment_397791 suggests we could try an OwnCloud install in Docker.
